src/useSsrData.ts

```diff
@@ -1,58 +1,82 @@
+
+
 export { SsrDataProvider }
 export { useSsrData }
 
 import React, { useContext } from 'react'
 import { useStream } from './useStream'
 import { assert, isClientSide, isServerSide } from './utils'
 import { parse, stringify } from '@brillout/json-s'
+import { DependencyList } from "./types";
 
 const Ctx = React.createContext<Data>(undefined as any)
 
 type Data = Record<string, Entry>
 type Entry =
-  | { state: 'pending'; promise: Promise<unknown> }
-  | { state: 'error'; error: unknown }
-  | { state: 'done'; value: unknown }
+  | { state: "pending"; promise: Promise<unknown>; deps?: DependencyList }
+  | { state: "error"; error: unknown }
+  | { state: "done"; value: unknown; deps?: DependencyList };
 
 function SsrDataProvider({ children }: { children: React.ReactNode }) {
   const data = {}
   return React.createElement(Ctx.Provider, { value: data }, children)
 }
 
-type SsrData = { key: string; value: unknown }
+type SsrData = { key: string; value: unknown; deps?: DependencyList };
 const className = 'react-streaming_ssr-data'
 function getHtmlChunk(entry: SsrData): string {
   const ssrData = [entry]
-  const htmlChunk = `<script class="${className}" type="application/json">${stringify(ssrData)}</script>`
-  return htmlChunk
+  return `<script class="${className}" type="application/json">${stringify(ssrData)}</script>`
 }
 
-function getSsrData(key: string): { isAvailable: true; value: unknown } | { isAvailable: false } {
-  const els = Array.from(window.document.querySelectorAll(`.${className}`))
+function getJsonScriptElement(
+  key: string
+): { el: Element; entry: SsrData } | undefined {
+  const els = Array.from(window.document.querySelectorAll(`.${className}`));
   for (const el of els) {
-    assert(el.textContent)
-    const data = parse(el.textContent) as SsrData[]
+    assert(el.textContent);
+    const data = parse(el.textContent) as SsrData[];
     for (const entry of data) {
-      assert(typeof entry.key === 'string')
+      assert(typeof entry.key === "string");
       if (entry.key === key) {
-        const { value } = entry
-        return { isAvailable: true, value }
+        return { el, entry };
       }
     }
   }
-  return { isAvailable: false }
+  return;
 }
 
-function useSsrData<T>(key: string, asyncFn: () => Promise<T>): T {
+function getSsrData(
+  key: string
+):
+  | { isAvailable: true; value: unknown; el: Element; deps?: DependencyList }
+  | { isAvailable: false } {
+  const { el, entry } = getJsonScriptElement(key) || {};
+
+  if (el && entry) {
+    const { value, deps } = entry;
+    return { isAvailable: true, value, deps, el };
+  }
+  return { isAvailable: false };
+}
+
+function useSsrData<T>(key: string, asyncFn: () => Promise<T>, deps?: DependencyList): T {
+  const data = useContext(Ctx);
+  let hasChanged = false;
   if (isClientSide()) {
     const ssrData = getSsrData(key)
     if (ssrData.isAvailable) {
-      return ssrData.value as T
+      if (deps) {
+        assert(ssrData?.deps)
```

Comment:
That's not true though, is it?

Reply:
You're right. I've removed it.

Reply:
Shouldn't `hasChanged` be `true` if either one of `ssrData.deps` and `deps` is missing while the other is not?

Reply:
What about that ?
```
if (deps || ssrData.deps) {
        hasChanged = true;
        if (deps &&  ssrData.deps) {
          hasChanged = ssrData?.deps.some(
          (d, index) => !Object.is(d, deps[index])
        );
        }
      }
```

Reply:
Pushed the changed. Thx